### PR TITLE
[Merged by Bors] - Constify SpritePipelineKey implementation.

### PR DESCRIPTION
**Pull request by AxiomaticSemantics**
# Objective

- Describe the objective or issue this PR addresses.
SpritePipelineKey could use more constification.

## Solution
Constify SpritePipelineKey implementation.

### CODE REVIEW

**Review by james7132 — APPROVED** [review]

File: crates/bevy_sprite/src/render/mod.rs [hunk 1]
```diff
@@ -160,25 +160,31 @@ impl SpritePipelineKey {
...
 
-    pub fn from_msaa_samples(msaa_samples: u32) -> Self {
+    #[inline]
+    pub const fn from_msaa_samples(msaa_samples: u32) -> Self {
         let msaa_bits =
             (msaa_samples.trailing_zeros() & Self::MSAA_MASK_BITS) << Self::MSAA_SHIFT_BITS;
-        Self::from_bits(msaa_bits).unwrap()
+        // SAFETY: The input is the correct type and the mask will ensure the output range is
+        // capped
+        unsafe { Self::from_bits_unchecked(msaa_bits) }
```
Comment (hymm) [inline crates/bevy_sprite/src/render/mod.rs:169]: For other reviewers here's the safety comment from the bitflags crate

https://docs.rs/bitflags/latest/src/bitflags/lib.rs.html#575-577

```rust
/// The caller of `from_bits_unchecked()` has to ensure that
/// all bits correspond to a defined flag or that extra bits
/// are valid for this bitflags type.
```

**Review by hymm — APPROVED** [review]
Could we add some docs for these functions? Not going to block on that though.

File: crates/bevy_sprite/src/render/mod.rs (hunk 1 above)
Comment (cart) [inline crates/bevy_sprite/src/render/mod.rs:169]: Idk if this is a good enough reason to introduce unsafe code. Does const-ing this unlock specific scenarios we need? Are there performance benefits?

Comment (james7132) [reply crates/bevy_sprite/src/render/mod.rs:169]:
```suggestion
        Self::from_bits_truncate(msaa_bits)
```
This should be `const`  and safe.

Comment (AxiomaticSemantics) [reply crates/bevy_sprite/src/render/mod.rs:169]: Thanks, I overlooked that, will fix it shortly.

Comment (AxiomaticSemantics) [reply crates/bevy_sprite/src/render/mod.rs]: Fixed.

**Review by james7132 — APPROVED** [review]

**Pull request closed without merging**